app/services/ucas_matching/process_matching_data.rb

```diff
@@ -88,6 +88,7 @@ def process_csv_file(file)
       candidates.each do |candidate_id, matching_data|
         match = UCASMatch.find_or_initialize_by(candidate_id: candidate_id)
         match.matching_data = matching_data
+        match.recruitment_cycle_year = RecruitmentCycle.current_year
```

Comment:
Should this be inside a block attached to `find_or_initialize_by`, just in case we accidentally overwrite the year on an existing match?

Reply:
I understand what the problem may be here but not sure what you mean by the block. 
Would something like this do it?
`match.recruitment_cycle_year = RecruitmentCycle.current_year if match.recruitment_cycle_year.nil?`

Reply:
If you pass a block to `find_or_create_by` or `find_or_initialize_by` it only runs if the record doesn't exist yet: https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by